batcher/batcher.mligo

```diff
@@ -349,9 +352,54 @@ module TokenAmountMap = struct
 
 end
 
-module Storage = struct
+module BatchHoldings_Utils = struct
+
+
+[@inline]
+let increase_holding
+  (batch_number: nat)
+  (batch_holdings: batch_holdings): batch_holdings =
+  match Big_map.find_opt batch_number batch_holdings with
+  | None -> Big_map.add batch_number 1n batch_holdings
+  | Some h -> let nh = h + 1n in
+              Big_map.update batch_number (Some nh) batch_holdings
+
+
+[@inline]
+let add_batch_holding
+  (batch_number: nat)
+  (address: address)
+  (ubots: user_batch_ordertypes)
+  (batch_holdings: batch_holdings): batch_holdings =
+  match Big_map.find_opt address ubots with
+  | Some bots -> (match Map.find_opt batch_number bots with
+                  | Some _ -> batch_holdings
+                  | None   -> increase_holding batch_number batch_holdings)
+  | None -> increase_holding batch_number batch_holdings
+
+[@inline]
+let remove_batch_holding
+  (batch_number: nat)
+  (batch_holdings: batch_holdings): batch_holdings =
+  match Big_map.find_opt batch_number batch_holdings with
+  | None -> failwith cannot_decrease_holdings_of_removed_batch
+  | Some h -> let nh = abs(h - 1n) in
+              if nh > 0n then
+                 Big_map.update batch_number (Some nh) batch_holdings
+              else
+                 Big_map.remove batch_number batch_holdings
+
+
+[@inline]
+let can_batch_be_removed
+  (batch_number: nat)
+  (batch_holdings: batch_holdings): bool = Big_map.mem batch_number batch_holdings
```

Comment:
I can remove a batch *if* it's in `batch_holdings`?

Reply:
Will fix